### PR TITLE
Zendesk Mobile: add support for Jetpack and self-hosted users.

### LINKED ISSUE #9170: Zendesk Mobile: add support for Jetpack and self-hosted users.
Enable Zendesk support features for Jetpack and self-hosted (no-JP) sites.

When not logged in with a WP account, and have self-hosted sites attached:
- When accessing Support the first time, save the user name and email from the last used site to User Defaults. Use that information for the current Support session.
- Use that information for all future Support options.
- This information will always be used unless the user re-installs the app or logs into a WP account.

The flow for selecting which user information to use:
1. If there is a WordPress.com account, use that.
2. If not, check if we’ve saved an email for Zendesk. If so, use that.
3. If there is no email saved, get the email from the selected site, save that, and use it for Zendesk.

### PR DESCRIPTION
**Pull request by ScoutHarris**
Fixes #9170 

**NOTES:**
- Your local `.wpcom_app_credentials` must be up-to-date as it contains Zendesk credentials. If you have not, refer to the Field Guide for instructions.
- Accessing 'Help' from Signup or Login is not working properly yet.
- There is a distinct lack of error handling. That's on my to-do list for later.

**To test:**
- Be sure you are _not_ logged in with a WP account.
- Connect multiple self-hosted sites. 
- Select any site.
  - Create a Support ticket.
  - Verify that ticket appears in the 'My Tickets' list.
- Select another site.
  - Verify that ticket still appears in the 'My Tickets' list.
  - Create another Support ticket.
  - Verify both tickets appears in the 'My Tickets' list.

In short, no matter which self-hosted site is selected, 'My Tickets' should show all tickets.

@jleandroperez - would you have time to review this? Thank you!

### CODE REVIEW

File: WordPress/Classes/Utility/ZendeskUtils.swift [hunk 1]
```diff
@@ -7,10 +7,13 @@ import CoreTelephony
     // MARK: - Properties
 
     static var sharedInstance: ZendeskUtils = ZendeskUtils()
+    private override init() {}
 
-    static var zendeskEnabled: Bool = false
+    var zendeskEnabled: Bool = false
```
Comment (jleandroperez) [inline WordPress/Classes/Utility/ZendeskUtils.swift]: Nipticky: no need to declare `: Bool`, since the type is inferred

File: WordPress/Classes/Utility/ZendeskUtils.swift [hunk 2]
```diff
@@ -179,31 +295,40 @@ private extension ZendeskUtils {
...
+            return [Constants.wpComTag]
+        }
 
-            tags.append(Constants.wpComTag)
+        // Get all unique site plans
+        var tags = allBlogs.filter {
+            $0.planTitle != nil
+            }.map {
+                $0.planTitle!
+            }.unique
```
Comment (jleandroperez) [inline WordPress/Classes/Utility/ZendeskUtils.swift]: (Unless i'm missing something!!) this should be equivalent to this snippet:

```
var tags = allBlogs.flatMap { $0.planTitle }.unique
```

`flatMap` should remove the `nil` values automatically for you. Super cool!

File: WordPress/Classes/Utility/ZendeskUtils.swift [hunk 3]
```diff
@@ -179,31 +295,40 @@ private extension ZendeskUtils {
...
 
-        } else {
+        // If any of the sites have jetpack installed, add jetpack tag.
+        let jetpackInstalled = allBlogs.map { $0.jetpack?.isInstalled }
+        if jetpackInstalled.contains(true) {
```
Comment (jleandroperez) [inline WordPress/Classes/Utility/ZendeskUtils.swift]: You can probably simplify this snippet like this:

```
let jetpackBlog = allBlogs.first { $0.jetapck?.isInstalled == true }
if let _ = jetpackBlog {
   // ..
}
```

This way you both, stop iterating thru the blogs whenever one of them is `jetpack enabled`, and avoid checking for `true` values into the array (the variable should either be nil, or have the first jetpack enabled blog).

File: WordPress/Classes/Utility/ZendeskUtils.swift [hunk 4]
```diff
@@ -25,34 +28,21 @@ import CoreTelephony
...
-        zendeskIdentity.email = accountSettings.email
-        zendeskIdentity.name = userName
-        ZDKConfig.instance().userIdentity = zendeskIdentity
-        ZendeskUtils.identityCreated = true
-    }
-
-    static func showHelpCenter(from controller: UIViewController) {
-
-        if !ZendeskUtils.identityCreated {
+        if !ZendeskUtils.sharedInstance.identityCreated {
```
Comment (jleandroperez) [inline WordPress/Classes/Utility/ZendeskUtils.swift]: Is this equivalent to `!identityCreated` ?

File: WordPress/Classes/Utility/ZendeskUtils.swift [hunk 5]
```diff
@@ -64,33 +54,33 @@ import CoreTelephony
...
-        let presentInController = ZendeskUtils.sharedInstance.configureViewController(controller)
+        let presentInController = ZendeskUtils.configureViewController(controller)
         ZDKHelpCenter.presentOverview(presentInController, with: helpCenterContentModel)
     }
 
-    static func showNewRequest(from controller: UIViewController) {
+    func showNewRequest(from controller: UIViewController) {
 
-        if !ZendeskUtils.identityCreated {
+        if !ZendeskUtils.sharedInstance.identityCreated {
```
Comment (jleandroperez) [inline WordPress/Classes/Utility/ZendeskUtils.swift]: Sames a above, is this equivalent to `if !identityCreated` ?

File: WordPress/Classes/Utility/ZendeskUtils.swift [hunk 6]
```diff
@@ -179,31 +295,40 @@ private extension ZendeskUtils {
...
-        } else {
+        // If any of the sites have jetpack installed, add jetpack tag.
+        let jetpackInstalled = allBlogs.map { $0.jetpack?.isInstalled }
+        if jetpackInstalled.contains(true) {
             tags.append(Constants.jetpackTag)
         }
 
+
+        // If there is a WP account, add wpcom tag.
+        let accountService = AccountService.init(managedObjectContext: context)
```
Comment (jleandroperez) [inline WordPress/Classes/Utility/ZendeskUtils.swift]: No need to explicitly call `init`:

```
let accountService = AccountService(managedObjectContext: context)
```

File: WordPress/Classes/Utility/ZendeskUtils.swift [hunk 7]
```diff
@@ -64,33 +54,33 @@ import CoreTelephony
...
-        let presentInController = ZendeskUtils.sharedInstance.configureViewController(controller)
+        let presentInController = ZendeskUtils.configureViewController(controller)
         ZDKRequests.presentRequestList(with: presentInController)
     }
 
-    static func createRequest() {
+    func createRequest() {
 
-        if !ZendeskUtils.identityCreated {
+        if !ZendeskUtils.sharedInstance.identityCreated {
```
Comment (jleandroperez) [inline WordPress/Classes/Utility/ZendeskUtils.swift]: Sames as above: is this equivalent to `if !identityCreated` ?

File: WordPress/Classes/Utility/ZendeskUtils.swift [hunk 8]
```diff
@@ -127,12 +116,69 @@ import CoreTelephony
 
 private extension ZendeskUtils {
 
-    func enableZendesk(_ enabled: Bool) {
-        ZendeskUtils.zendeskEnabled = enabled
+    static func enableZendesk(_ enabled: Bool) {
```
Comment (jleandroperez) [inline WordPress/Classes/Utility/ZendeskUtils.swift]: Nipticky suggestion!

`enableZendesk(false)` actually disables Zendesk, but that's not clear by just looking at the method signature.

How about.... 

- `static func toggleZendesk(enabled: Bool)` 
- `static func configureZendesk(enabled: Bool)`
- `static func enableZendesk()` + `static fund disableZendesk()`

**Comment by ScoutHarris** [comment]
@jleandroperez - thanks for the great feedback! I've addressed them all. One more time please.
